epan/dissectors/packet-ippusb.c

```diff
@@ -25,66 +25,104 @@
 #include "packet-usb.h"
 #include <stdio.h>
 
+#define HTTP 0
+#define IPP 1
+
 void proto_register_ippusb(void);
 void proto_reg_handoff_ippusb(void);
+static int is_http_header(int first_linelen, const guchar *first_line); 
 
 static int proto_ippusb = -1;
+static gint ett_ippusb = -1;
+static gint ett_ippusb_as = -1;
+static gint ett_ippusb_attr = -1;
+static gint ett_ippusb_member = -1;
+
 static dissector_table_t ippusb_dissector_table;
 
 static int
 dissect_ippusb(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, void* data)
 {
     //skip to http header info
     //put dummy info in for http dissector
-
-    proto_tree  *ippusb_tree = tree;
-    //proto_item  *ti;
-    int         offset     = 0;
-    //http_message_info_t *message_info = (http_message_info_t *)data;
-    //gboolean    is_request;
-    //guint16     operation_status;
-   // const gchar *status_type;
-    //guint32	request_id;
-    //conversation_t *conversation;
-
-    //offset +=
-
-    if (tvb_captured_length_remaining(tvb, offset) > 0) {
-        next_tvb = tvb_new_subset_remaining(tvb, offset);
-        offset += try_dissect_next_protocol(parent, next_tvb, pinfo, usb_conv_info, urb_type, tree, NULL);
+    proto_tree  *ippusb_tree;
+    proto_item  *ti;
+    int offset = 0;
+    int ret;
+    int first_linelen;
+    const guchar *first_line;
+    gint next_offset;
+
+    ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+    ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+    first_linelen = tvb_find_line_end(tvb, offset,  tvb_ensure_captured_length_remaining(tvb, offset), &next_offset, TRUE);
+    first_line = tvb_get_ptr(tvb, offset, first_linelen);
+
+    //check if there is a http header to parse before ipp
+    if (is_http_header(first_linelen, first_line)) {        
```

Comment:
>    if (is_http_header(first_linelen, first_line)) {   [](start = 0, length = 54)

David and Michael - Jamie and I chatted about this and the HTTP header is only available sometimes. If that is the case do we want to ever pass it to the HTTP dissector? I was just thinking it might be weird for a user of wireshark to sometimes see the HTTP header in the tree and sometimes not see it. 

It appears I cant @ individuals in this version of codeflow.

Reply:
Discussed offline. 

---
In reply to: [448652503](https://github.com/tjamhar/wireshark/pull/2#discussion_r448652503) [](ancestors = 448652503)

Reply:
So what hasn't been done yet is getting the USB continuation frames to coalesce with the main frame that begins with HTTP.  That should address the concern once completed.

---
In reply to: [448674928](https://github.com/tjamhar/wireshark/pull/2#discussion_r448674928) [](ancestors = 448674928,448652503)